src/lib/db/group-store.ts

```diff
@@ -210,10 +210,9 @@ export default class GroupStore implements IGroupStore {
         deletableUsers: IGroupUser[],
         userName: string,
     ): Promise<void> {
-        await this.db.transaction(async (tx) => {
-            await this.addUsersToGroup(groupId, newUsers, userName, tx);
-            await this.deleteUsersFromGroup(deletableUsers, tx);
-        });
+        this.expectTransaction(this.db);
```

Comment:
This got my attention. Does it make sense for a store to require the caller to wrap the call in a transaction? What would happen if we left the original code as is and the service wrapped the transaction in another transaction?

Reply:
Maybe this method does not belong in the store anymore, if we know move the transaction handling to the service?

Reply:
This is a method we added in this PR, in my mind this was less about stores telling services things and more about programmers in the past telling programmers in the future things. What I wanted to try and avoid here is folks making messy store methods that require multiple calls without a way to signal to future developers that it's not safe to be run in a non transactional context. This shouldn't do anything outside of a test harness - this is really just a way to make people's tests crash hard if they do silly things

I... actually haven't tried to wrap a transaction in a child transaction. There's an open issue on Knex that suggests that deadlocks but let me try it and see what happens.